lib/ansible/plugins/action/unarchive.py

```diff
@@ -38,9 +38,20 @@ def run(self, tmp=None, task_vars=None):
 
         source  = self._task.args.get('src', None)
         dest    = self._task.args.get('dest', None)
-        copy    = boolean(self._task.args.get('copy', True))
+        remote_src = boolean(self._task.args.get('remote_src', False))
         creates = self._task.args.get('creates', None)
 
+        # "copy" is deprecated in favor of "remote_src".
+        if 'copy' in self._task.args:
+            # They are mutually exclusive.
+            if 'remote_src' in self._task.args:
+                result['failed'] = True
+                result['msg'] = "parameters are mutually exclusive: ('copy', 'remote_src')"
+                return result
+            # We will take the information from copy and store it in
+            # the remote_src var to use later in the playbook
```

Comment:
> Right, and the same should logic should be used in the unarchive module IMO.

what logic specifically? I tried to match what the unarchive module is doing. From a users perspective they can provide either `copy` or `remote_src` or neither of them and it should work as desired. The logic is consistent between both. 

> 
> I was wondering whether we should fail for mutual exclusive options (when deprecated), or simply take the non-deprecated option and throw a warning to the user. Would be nice if Ansible does this consistently everywhere options are being deprecated.

I decided to fail because that is what the module does already: https://github.com/ansible/ansible-modules-core/blob/91a839f1e3de58be8b981d3278aa5dc8ff59c508/files/unarchive.py#L721